indico/modules/events/registration/templates/management/reglist_filter.html

```diff
@@ -31,7 +31,7 @@
 
 {% macro _render_regform_item_col_selector(section) %}
     <h3>{{ section.title }}</h3>
-    {% for child_item in section.children if child_item.is_field and not child_item.is_deleted %}
+    {% for child_item in section.children if child_item.is_field and not (child_item.is_deleted or child_item.parent.is_deleted) %}
```

Comment:
Actually. Let's not change this line at all but simply adapt `{% for section in regform.sections %}` when the macro is called and skip deleted sections there.